packages/compass-databases-navigation/src/databases-navigation-tree.tsx

```diff
@@ -142,14 +142,18 @@ const DatabasesNavigationTree: React.FunctionComponent<{
   onDatabaseExpand(id: string, isExpanded: boolean): void;
   onNamespaceAction(namespace: string, action: Actions): void;
   activeNamespace?: string;
-  isReadOnly?: boolean;
```

Comment:
I would have a preference for this to be fixed on the usage side instead of worked around here and keep the interface of the component as straightforward as possible: we are adding these three new props and they basically just resolve to calculating one value at the very top level of the component to be passed down to all the other parts of the component, now you need to know about all three and what they mean, even though from component perspective we only care about one value that should just hide the button. Does this make sense to you? 🤔